unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/Scripts/NFTShapeAssetsType.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System;
+using System.Collections;
+using DCL;
+using DCL.Controllers.Gif;
+using UnityEngine;
+
+namespace NFTShape_Internal
+{
+    public interface INFTAsset : IDisposable
+    {
+        bool isHQ { get; }
+        int hqResolution { get; }
+        DCL.ITexture previewAsset { get; }
+        DCL.ITexture hqAsset { get; }
+        Action<Texture2D> UpdateTextureCallback { set; }
+        void FetchAndSetHQAsset(string url, Action onSuccess, Action onFail);
+        void RestorePreviewAsset();
+    }
+
+    public static class NFTAssetFactory
+    {
+        public static INFTAsset CreateAsset(ITexture asset, NFTShapeConfig shapeConfig)
+        {
+            if (asset == null)
+            {
+                return null;
+            }
+
+            if (asset is Asset_Gif gif)
+            {
+                return new GifAsset(gif, shapeConfig.hqGifResolution);
+            }
+
+            return new ImageAsset(asset, shapeConfig.hqImgResolution);
+        }
+    }
+
+    public class ImageAsset : INFTAsset
```

Comment:
Why are all those classes under a single file? We can't create individual files for them? It'd easier for VCS.

We could do the following:
- Remove `NFTShapeAssetsType`
- Add `NFTImageAsset.cs`
- Add `NFTGifAsset.cs`
- Add `INFTAsset.cs`

Reply:
Cause I'm super lazy! but you are totally right, now that I'm refactoring I could take the time to do it the right way :P

Reply:
done